app/services/blood_pressure_export_service.rb

```diff
@@ -0,0 +1,169 @@
+class BloodPressureExportService
+  require "csv"
+  include ActionView::Helpers::NumberHelper
+  include DashboardHelper
+
+  attr_reader :start_period, :end_period, :facilities,  :data_for_facility, :stats_by_size, :display_sizes
+
+  FACILITY_SIZES = %w[large medium small community]
+  DATA_TYPES = [:controlled_patients_rate, :uncontrolled_patients_rate, :missed_visits_rate]
+
+  def initialize(start_period:, end_period:, facilities:)
+    @start_period = start_period
+    @end_period = end_period
+    @facilities = facilities
+
+    presenter = Reports::RepositoryPresenter.create(facilities, period: @end_period, months: 6, reporting_schema_v2: RequestStore.store[:reporting_schema_v2])
+    @data_for_facility = facilities.each_with_object({}) do |facility, result|
+      result[facility.name] = presenter.my_facilities_hash(facility.region)
+    end
+    found_sizes = @facilities.pluck(:facility_size).uniq
+    @sizes = FACILITY_SIZES.select { |size| size.in?(found_sizes) }
+    @stats_by_size = FacilityStatsService.call(facilities: @data_for_facility, period: @end_period)
+  end
+
+  def call
+    aggregate_data
+  end
+
+  def as_csv
+    CSV.generate {|csv|
+      headers = set_csv_headers
+      headers.each do |header_row|
+        csv << header_row
+      end
+      data = aggregate_data
+      data.keys.each_with_index do |size, i|
+        aggregate = data[size]["aggregate"]
+        row = []
+        row << aggregate["Facilities"]
+        row << aggregate["Total assigned"]
+        row << aggregate["Total registered"]
+        aggregate[:controlled_patients_rate].each_pair do |key,value|
+          row << value
+        end
+        aggregate[:uncontrolled_patients_rate].each_pair do |key,value|
+          row << value
+        end
+        aggregate[:missed_visits_rate].each_pair do |key,value|
+          row << value
+        end
+
+        csv << row
+        facilities = data[size]["facilities"]
+        facilities.each do |facility|
+          facility_row = []
+          facility_row << facility["Facilities"]
+          facility_row << facility["Total assigned"]
+          facility_row << facility["Total registered"]
+          DATA_TYPES.each do |rate_type|
+            facility[rate_type].each_pair do |key, value|
+              facility_row << value
+            end
+          end
+          csv << facility_row
+        end
+        csv << [] if i != @sizes.length - 1
+      end
+    }
+  end
+
+  private
+
+  def aggregate_data
+    @aggregate_data ||= begin
+      formatted = {}
+      @sizes.each do |size|
+        if !formatted[size]
+          formatted[size] = {}
+          formatted[size]["aggregate"] = format_aggregate_facility_stats(size)
+          formatted[size]["facilities"] = format_facilities_of_size(size)
+        end
+      end
+      formatted
+    end
+  end
+
+  def set_csv_headers
+    headers = []
+    spacing = (@start_period..@end_period).map{""}
```

Comment:
Looks like some linting things going on here -- I would recommend running `script/lint.sh` and committing that to clean them up.